internal/xds/runner/runner.go

```diff
@@ -96,11 +96,41 @@ func (r *Runner) Name() string {
 	return string(egv1a1.LogComponentXdsRunner)
 }
 
-func defaultServerKeepaliveParams() keepalive.ServerParameters {
-	return keepalive.ServerParameters{
+func (r *Runner) serverKeepaliveParams() (keepalive.ServerParameters, error) {
+	params := keepalive.ServerParameters{
 		MaxConnectionAge:      getRandomMaxConnectionAge(),
 		MaxConnectionAgeGrace: defaultMaxConnectionAgeGrace,
 	}
+
+	if r.EnvoyGateway == nil || r.EnvoyGateway.XDSServer == nil {
+		return params, nil
+	}
+
+	cfg := r.EnvoyGateway.XDSServer
+
+	if cfg.MaxConnectionAge != nil {
+		d, err := time.ParseDuration(string(*cfg.MaxConnectionAge))
+		if err != nil {
+			return keepalive.ServerParameters{}, fmt.Errorf("invalid xdsServer.maxConnectionAge: %w", err)
+		}
+		if d <= 0 {
+			return keepalive.ServerParameters{}, fmt.Errorf("xdsServer.maxConnectionAge must be greater than zero")
+		}
+		params.MaxConnectionAge = d
```

Comment:
Maybe we should also add some randomness to d so the envoy fleet don't reconnect to the control plane at the same time?